zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -117,38 +121,36 @@ pub(crate) fn get_transparent_receivers<P: consensus::Parameters>(
     ])?;
 
     while let Some(row) = rows.next()? {
-        let ua_str: String = row.get(0)?;
-        let di_vec: Vec<u8> = row.get(1)?;
+        let addr_str: String = row.get(0)?;
+        let address_index: u32 = row.get(1)?;
+        let address_index = NonHardenedChildIndex::from_index(address_index).ok_or(
+            SqliteClientError::CorruptedData(format!(
+                "{} is not a valid transparent child index",
+                address_index
+            )),
+        )?;
         let scope = KeyScope::decode(row.get(2)?)?;
 
-        let taddr = Address::decode(params, &ua_str)
+        let taddr = Address::decode(params, &addr_str)
             .ok_or_else(|| {
                 SqliteClientError::CorruptedData("Not a valid Zcash recipient address".to_owned())
             })?
             .to_transparent_address();
```

Comment:
Does this do the right thing when the decoded address is already a t-address? It should; just checking.

Reply:
Yes; it returns the p2pkh address for bare taddrs, TEX addresses, and UAs with transparent receivers.